pkg/executor/statefulset.go

```diff
@@ -273,18 +274,35 @@ func getStatefulset(
 		return nil, securityErr
 	}
 
-	return &appsv1.StatefulSet{
-		TypeMeta: metav1.TypeMeta{
-			Kind:       "StatefulSet",
-			APIVersion: "apps/v1",
-		},
-		ObjectMeta: metav1.ObjectMeta{
+	namingScheme := *cr.Spec.NamingScheme
+	var objectName string
+	var objectMeta metav1.ObjectMeta
+	if namingScheme == v1beta1.CrNameRole {
+		objectName = ValidateName(cr.Name + "-" + role.Name)
+		objectName += "-"
+		objectMeta = metav1.ObjectMeta{
+			GenerateName:    objectName,
```

Comment:
What's the need for using GenerateName in this scheme, as opposed to just Name?

Reply:
You are correct, I dont think we need to use generate names here.